### PR TITLE
[WIP][GCS]minimize size of actor table data in publishing

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?

Currently we publish whole actor table data when a new actor registered or any actor's states changed, which carries lots of redundant information as size of ActorTableData is pretty large.
We can publish actor's states(state, address, pid, num_restarts, timestamp) only, for reducing redis pressure.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
I will review this tomorrow (sorry for the delay)

File: src/ray/core_worker/actor_manager.h [hunk 1]
```diff
@@ -175,9 +175,9 @@ class ActorManager {
   /// Handle actor state notification published from GCS.
   ///
   /// \param[in] actor_id The actor id of this notification.
-  /// \param[in] actor_data The GCS actor data.
-  void HandleActorStateNotification(const ActorID &actor_id,
-                                    const rpc::ActorTableData &actor_data);
+  /// \param[in] actor_states The GCS actor states.
+  void HandleActorStatesNotification(const ActorID &actor_id,
+                                     const rpc::ActorStates &actor_states);
```
Comment (ffbin) [inline src/ray/core_worker/actor_manager.h:180]: `actor_state`?

File: src/ray/protobuf/gcs.proto [hunk 2]
```diff
@@ -94,7 +94,7 @@ message TaskTableData {
   Task task = 1;
 }
 
-message ActorTableData {
+message ActorStates {
```
Comment (ffbin) [inline src/ray/protobuf/gcs.proto:97]: `ActorStates` seems to be the plural of `ActorState`, which is confusing.

Comment (WangTaoTheTonic) [reply src/ray/protobuf/gcs.proto:97]: `ActorStates` including several states of actor, including state, timestamp, address, etc. It kinda confusing and better naming is welcome!

**Comment by WangTaoTheTonic** [comment]
I found this solution is kinda complicated. We can just set `states related field`, but not extract them from original data structure, like what we did to node publish in https://github.com/ray-project/ray/pull/13364
So I convert this to WIP status and welcome to review after new solution pushed.

**Comment by WangTaoTheTonic** [comment]
A simplified version is here https://github.com/ray-project/ray/pull/13448, I am closing this. thanks!

**Pull request closed without merging**